pkg/instancetype/preference/find/spec_test.go

```diff
@@ -0,0 +1,250 @@
+	BeforeEach(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubecli.NewMockKubevirtClient(ctrl)
+
+		virtClient.EXPECT().AppsV1().Return(k8sfake.NewSimpleClientset().AppsV1()).AnyTimes()
+
+		fakeClientset = fake.NewSimpleClientset()
+
+		virtClient.EXPECT().VirtualMachine(metav1.NamespaceDefault).Return(
+			fakeClientset.KubevirtV1().VirtualMachines(metav1.NamespaceDefault)).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachineClusterPreference().Return(
+			fakeClientset.InstancetypeV1beta1().VirtualMachineClusterPreferences()).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachinePreference(metav1.NamespaceDefault).Return(
+			fakeClientset.InstancetypeV1beta1().VirtualMachinePreferences(metav1.NamespaceDefault)).AnyTimes()
+
+		preferenceInformer, _ := testutils.NewFakeInformerFor(&v1beta1.VirtualMachinePreference{})
+		preferenceInformerStore = preferenceInformer.GetStore()
+
+		clusterPreferenceInformer, _ := testutils.NewFakeInformerFor(&v1beta1.VirtualMachineClusterPreference{})
+		clusterPreferenceInformerStore = clusterPreferenceInformer.GetStore()
+
+		controllerRevisionInformer, _ := testutils.NewFakeInformerFor(&appsv1.ControllerRevision{})
+		controllerRevisionInformerStore = controllerRevisionInformer.GetStore()
+
+		finder = find.NewSpecFinder(preferenceInformerStore, clusterPreferenceInformerStore, controllerRevisionInformerStore, virtClient)
+	})
+
+	It("find returns nil when no preference is specified", func() {
+		vm = libvmi.NewVirtualMachine(libvmi.New())
+		preference, err := finder.FindPreference(vm)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(preference).To(BeNil())
+	})
+
+	It("find returns error when invalid Preference Kind is specified", func() {
+		vm = libvmi.NewVirtualMachine(libvmi.New())
+		vm.Spec.Preference = &v1.PreferenceMatcher{
+			Name: "foo",
+			Kind: "bar",
+		}
+		spec, err := finder.FindPreference(vm)
+		Expect(err).To(MatchError(ContainSubstring("got unexpected kind in PreferenceMatcher")))
+		Expect(spec).To(BeNil())
+	})
+
+	Context("Using global ClusterPreference", func() {
+		var clusterPreference *v1beta1.VirtualMachineClusterPreference
+
+		BeforeEach(func() {
+			preferredCPUTopology := v1beta1.Cores
+			clusterPreference = &v1beta1.VirtualMachineClusterPreference{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-cluster-preference",
+				},
+				Spec: v1beta1.VirtualMachinePreferenceSpec{
+					CPU: &v1beta1.CPUPreferences{
+						PreferredCPUTopology: &preferredCPUTopology,
+					},
+				},
+			}
+
+			_, err := virtClient.VirtualMachineClusterPreference().Create(context.Background(), clusterPreference, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			err = clusterPreferenceInformerStore.Add(clusterPreference)
+			Expect(err).ToNot(HaveOccurred())
+
+			vm = libvmi.NewVirtualMachine(libvmi.New(), libvmi.WithClusterPreference(clusterPreference.Name))
+		})
+
+		It("find returns expected preference spec", func() {
+			s, err := finder.FindPreference(vm)
```

Comment:
```suggestion
			preferenceSpec, err := finder.FindPreference(vm)
```

nit: for consistency